src/main/webapp/app/shared/http/file-uploader.service.ts

```diff
@@ -49,6 +49,42 @@ export class FileUploaderService {
         return this.http.post<FileUploadResponse>(url, formData).toPromise();
     }
 
+    /**
+     * Function which uploads a file. It checks for supported file extensions and file size.
+     * Options must be passed as a dictionary. E.g: { keepFileName: true }
+     * @async
+     * @param {Blob | File} file
+     * @param {string} fileName
+     * @param options
+     */
+    uploadMarkdownFile(file: Blob | File, fileName?: string, options?: any): Promise<FileUploadResponse> {
+        /** Check file extension **/
+        const fileExtension = fileName ? fileName.split('.').pop()!.toLocaleLowerCase() : file['name'].split('.').pop().toLocaleLowerCase();
+        if (this.acceptedFileExtensions.split(',').indexOf(fileExtension) === -1) {
+            return Promise.reject(
+                new Error(
```

Comment:
Will the error be passed to the alert service?

Reply:
Yes